rpcclient/chain.go

```diff
@@ -1224,3 +1238,24 @@ func (c *Client) GetBlockStatsAsync(hashOrHeight interface{}, stats *[]string) F
 func (c *Client) GetBlockStats(hashOrHeight interface{}, stats *[]string) (*btcjson.GetBlockStatsResult, error) {
 	return c.GetBlockStatsAsync(hashOrHeight, stats).Receive()
 }
+
+// Receive waits for the response promised by the future and returns the hash of
+// the best block in the longest block chain.
+func (r FutureGetBulkResult) Receive() (BulkResult, error) {
+	m := make(BulkResult)
+	res, err := receiveFuture(r)
+	if err != nil {
+		return nil, err
+	}
+	var arr []IndividualBulkResult
```

Comment:
`res` may have a different structure. Example:

```
{"result":null,"error":{"code":-32700,"message":"Parse error"},"id":null}
```

Reply:
wouldn't this be: `[{"result":null,"error":{"code":-32700,"message":"Parse error"},"id":null}]`?

Reply:
I don't remember how I encountered it, but it was during some tests. I can try to reproduce it again.